vm/src/vm.rs

```diff
@@ -416,7 +416,9 @@ impl VirtualMachine {
 
     #[cfg(feature = "rustpython-compiler")]
     pub fn new_syntax_error(&self, error: &CompileError) -> PyObjectRef {
-        let syntax_error_type = if error.is_tab_error() {
+        let syntax_error_type = if error.is_indentation_error() {
```

Comment:
Instead of using a sequence of `if` statements, we could use here `match` statements on the `error` variable to check all it's variants.

Reply:
The main motivation of current design is to simplify compiler/VM public interface. In a sense {CompilerErrorType, ParseErrorType, LexicalErrorType} really are internal implementation details, the only thing VM needs is exception type, message, and location.

Reply:
I understand the motivation, but I still feel, the vm is so much connected to the parser, it is actually fine to leak the internals of the parser into the vm code.

In any case, let's go with the current design.